### PR TITLE
Review, fix and update some documentations for Flutter SDK.

### PR DESCRIPTION
**Pull request by salihgueler**
TLDR: Overall, Flutter code enhancements, fixes and updates. Also fixes https://github.com/aws-amplify/docs/issues/4394

---------

### Longer version

For learning about the Flutter libraries and what is offered to the users, I went through all the documents and put together a document:

# Feedback on AWS Amplify Flutter Documentation

**Overall**: 
- Sometimes the images are not readable
- The code was outdated and did not have a consistent style

## [Getting Started](https://docs.amplify.aws/start/q/integration/flutter/) (https://docs.amplify.aws/start/q/integration/flutter/)

Prerequisities

1. On this [page](https://docs.amplify.aws/start/getting-started/installation/q/integration/flutter/#option-2-follow-the-instructions) (https://docs.amplify.aws/start/getting-started/installation/q/integration/flutter/#option-2-follow-the-instructions):

* the *Once you're signed in, Amplify CLI will ask you to create an IAM user* sentence missing the fact that, once people are logged in, they need to go back to the CLI.
* the GIF and the information does not match. GIF needs to be updated to select AdministratorAccess-Amplify.
* once the user is created, we should warn user to not lose their keys.

Set up fullstack project

1. Project creation is too basic, we should have at least provided organisation name or with resources to teach them about proper Flutter project creation
2. ([Regarding page](https://docs.amplify.aws/start/getting-started/setup/q/integration/flutter/#update-target-ios-platform) (https://docs.amplify.aws/start/getting-started/setup/q/integration/flutter/#update-target-ios-platform)) Why do we define the iOS and SDK version here but assign the Kotlin version on the previous page? That does not feel right.

Generate model files

1. Isn't it too early to generate GraphQL files for Getting Started guide? We seem to be throwing a lot of information to the people here and that might scare them off.

1. I think setting up the application and creating the first application can be two different tutorials. Pre-requisities and setting up project should be available for all documents/projects. Now it feels like, I need to have Amplify Studio to create an app. But I do not need it. 

I will skip the rest of the steps because they are highly couple to Amplify Studio. 

# [Libraries](https://docs.amplify.aws/lib/q/platform/flutter/) (https://docs.amplify.aws/lib/q/platform/flutter/)

[Project Setup](https://docs.amplify.aws/lib/project-setup/prereq/q/platform/flutter/) (https://docs.amplify.aws/lib/project-setup/prereq/q/platform/flutter/)

[Prerequisities](https://docs.amplify.aws/lib/project-setup/prereq/q/platform/flutter/) (https://docs.amplify.aws/lib/project-setup/prereq/q/platform/flutter/)

1. Same prerequisities feedback can be applied here as well.

[Create your application](https://docs.amplify.aws/lib/project-setup/create-application/q/platform/flutter/) (https://docs.amplify.aws/lib/project-setup/create-application/q/platform/flutter/)

1. Why do we add the analytics library already? It has its own category. 
2. The pubspec code is inconsistent between files.
3. Again the iOS version is 11 here but at the Getting Started guide we asked to assign it to 13, which one do we need to have? 
4. If we set the minSdkVersion and iOS version here, why do we have it as prerequisite?
5. [At this title](http://localhost:3000/lib/project-setup/create-application/q/platform/flutter/#3-provision-the-backend-with-amplify-cli) (http://localhost:3000/lib/project-setup/create-application/q/platform/flutter/#3-provision-the-backend-with-amplify-cli), `amplify init` command does not provide the options, it first suggests smart options and if you say no, then it shows the following.

[Null Safety](https://docs.amplify.aws/lib/project-setup/null-safety/q/platform/flutter/) (https://docs.amplify.aws/lib/project-setup/null-safety/q/platform/flutter/)

* Do we still need it? 

[Escape Hatch](http://localhost:3000/lib/project-setup/escape-hatch/q/platform/flutter/#signing-a-request) (http://localhost:3000/lib/project-setup/escape-hatch/q/platform/flutter/#signing-a-request)

* The code was not functional because it did not work as a dart program

[Analytics](http://localhost:3000/lib/analytics/getting-started/q/platform/flutter) (http://localhost:3000/lib/analytics/getting-started/q/platform/flutter)

We don't have pinpoint for push notifications yet, right? 

[Getting Started](https://docs.amplify.aws/lib/analytics/getting-started/q/platform/flutter/) (https://docs.amplify.aws/lib/analytics/getting-started/q/platform/flutter/)

1. If the next step is *Record Events* why do we have the same title here?

[API (GraphQL)](http://localhost:3000/lib/graphqlapi/getting-started/q/platform/flutter/) (http://localhost:3000/lib/graphqlapi/getting-started/q/platform/flutter/)

1. Generated files feels like Java class files than Dart files.

[Subscribe to Data](http://localhost:3000/lib/graphqlapi/subscribe-data/q/platform/flutter/#setup-subscription-with-callbacks) (http://localhost:3000/lib/graphqlapi/subscribe-data/q/platform/flutter/#setup-subscription-with-callbacks)

The first code example feels totally wrong. Dart language has some features that is doing exactly what is aimed there.  I will take care of it but wanted to state here.

[API (REST)](https://docs.amplify.aws/lib/restapi/fetch/q/platform/flutter/) (https://docs.amplify.aws/lib/restapi/fetch/q/platform/flutter/)

1. RestOptions body field feels wrong. Why are we expected to send out a list insteade of regular bodies with http calls?

[Authentication](http://localhost:3000/lib/auth/auth-events/q/platform/flutter/) (http://localhost:3000/lib/auth/auth-events/q/platform/flutter/)

[Auth Events](http://localhost:3000/lib/auth/auth-events/q/platform/flutter/) (http://localhost:3000/lib/auth/auth-events/q/platform/flutter/)

1. Can we find a list of authentication events?

[Datastore](https://docs.amplify.aws/lib/datastore/getting-started/q/platform/flutter/) (https://docs.amplify.aws/lib/datastore/getting-started/q/platform/flutter/)

[Getting Started](https://docs.amplify.aws/lib/datastore/getting-started/q/platform/flutter/) (https://docs.amplify.aws/lib/datastore/getting-started/q/platform/flutter/)

1. The init of the amplify feels really out of nowhere and it might be good to either remove it or put more clarity of the steps there
2. Should we keep "Coming soon" signs?
3. Exception types overall needs to be addressed

[Storage](https://docs.amplify.aws/lib/storage/getting-started/q/platform/flutter/) (https://docs.amplify.aws/lib/storage/getting-started/q/platform/flutter/)

1. The storage overall uses the image_picker and path_provider, should we add them as pre-requirements?

### CODE REVIEW

**Review by rachelnabors — APPROVED** [review]
Love it. This is a much needed update to our Flutter documentation. AWESOME!

**Review by dnys1 — APPROVED** [review]
So much good stuff here. Thanks for the hard work! 

Left some comments based on our discussions offline.

File: src/fragments/lib/datastore/flutter/conflict.mdx [hunk 1]
```diff
@@ -14,31 +14,37 @@ Finally you can configure the number of records to sync as an upper bound on ite
...
+        }
+
+        return ConflictResolutionDecision.applyRemote();
+      },
+      // Sync configuration defaults:
+      syncInterval: 86400,
+      syncMaxRecords: 10000,
+      syncPageSize: 1000,
+    );
+    await Amplify.addPlugin(datastorePlugin);
```
Comment (HuiSF) [inline src/fragments/lib/datastore/flutter/conflict.mdx:45]: Meant to add `await Amplify.configure()` call here as well?

File: src/fragments/lib/datastore/flutter/data-access/delete-snippet.mdx [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
-  } on DataStoreException catch (e) {
-    print('Delete failed: $e');
+Future<void> deletePostsWithId() async {
+  final oldPosts = await Amplify.DataStore.query(
+    Post.classType,
+    where: Post.ID.eq('123'),
+  );
+  for (final oldPost in oldPosts) {
+    try {
+      await Amplify.DataStore.delete(oldPost);
```
[the remark this answers is not in the document]
Comment (HuiSF) [reply src/fragments/lib/datastore/flutter/data-access/delete-snippet.mdx:11]: nit: The example query predicate ensures return 1 element with the result list, maybe unnecessary to demo a for loop here.

File: src/fragments/lib/datastore/flutter/data-access/query-predicate-multiple-snippet.mdx [hunk 3]
```diff
@@ -1,4 +1,9 @@
 ```dart
-List<Post> posts = await Amplify.DataStore.query(Post.classType,
-    where: Post.RATING.eq(2).and(Post.STATUS.eq(PostStatus.PUBLISHED)));
+Future<void> fetchPublishedPostsMoreThanFourRating() async {
```
Comment (HuiSF) [inline src/fragments/lib/datastore/flutter/data-access/query-predicate-multiple-snippet.mdx:2]: nit: `fetchPublishedWithRatingTwoPosts`?

File: src/fragments/lib/datastore/flutter/data-access/query-predicate-or-snippet.mdx [hunk 4]
```diff
@@ -1,4 +1,9 @@
 ```dart
-List<Post> posts = await Amplify.DataStore.query(Post.classType,
-    where: Post.RATING.eq(2).or(Post.STATUS.eq(PostStatus.PUBLISHED)));
+Future<void> fetchPublishedOrMoreThanFourRatingPosts() async {
```
Comment (HuiSF) [inline src/fragments/lib/datastore/flutter/data-access/query-predicate-or-snippet.mdx:2]: nit: `fetchPublishedOrWithRatingTwoPosts`?

File: src/fragments/lib/datastore/flutter/relational/delete-snippet.mdx [hunk 5]
```diff
@@ -1,7 +1,13 @@
 ```dart
-(await Amplify.DataStore.query(Post.classType, where: Post.ID.eq('123')))
-    .forEach((element) async {
-  await Amplify.DataStore.delete(element);
-  print('Deleted a post');
-});
+Future<void> deletePostWithId(String id) async {
+  final postsWithId = await Amplify.DataStore.query(
+    Post.classType,
+    where: Post.ID.eq(id),
```
Comment (HuiSF) [inline src/fragments/lib/datastore/flutter/relational/delete-snippet.mdx:5]: nit: This predicate ensure returns an empty list or a list has only one element, the following for loop in the code example may be unnecessary.

File: src/fragments/lib/datastore/flutter/relational/query-snippet.mdx [hunk 6]
```diff
@@ -1,8 +1,11 @@
 In order to retrieve all models that are related to a parent model, you can use query predicates with the query API:
 
 ```dart
-List<Comment> comments = await Amplify.DataStore.query(
+Future<void> fetchAllCommentsForParentId(String parentId) async {
```
Comment (HuiSF) [inline src/fragments/lib/datastore/flutter/relational/query-snippet.mdx:4]: nit:
```suggestion
Future<void> fetchAllCommentsByPostId(String postId) async {
```

File: src/fragments/lib/datastore/flutter/relational/query-snippet.mdx [hunk 7]
```diff
@@ -1,8 +1,11 @@
...
+  final comments = await Amplify.DataStore.query(
     Comment.classType,
-    where: Comment.POST.eq('a-post-id'),
-);
-```
\ No newline at end of file
+    where: Comment.POST.eq(parentId),
```
Comment (HuiSF) [inline src/fragments/lib/datastore/flutter/relational/query-snippet.mdx:7]: nit:
```suggestion
    where: Comment.POST.eq(postId),
```

File: src/fragments/lib/datastore/flutter/sync/50-selectiveSync.mdx [hunk 8]
```diff
@@ -8,15 +8,23 @@ You can utilize selective sync to only persist a subset of your data instead.
...
+        () => Comment.POST.beginsWith('This'),
+      )
+    ],
+  );
+  ...
+}
 ```
 
-When DataStore starts syncing, only Posts with `rating > 5` and Comments with `status` equal to `active` will be synced down to the user's local store.
+When DataStore starts syncing, only Posts with `rating > 5` and Comments with `POST` id begins with `This` will be synced down to the user's local store.
```
Comment (HuiSF) [inline src/fragments/lib/datastore/flutter/sync/50-selectiveSync.mdx:27]: nit:
```suggestion
When DataStore starts syncing, only Posts with `rating > 5` and Comments with `POST` id begins with `"This"` will be synced down to the user's local store.
```

**Review by HuiSF — COMMENTED** [review]
Thanks for fixing the documents up!

File: src/fragments/lib/datastore/flutter/data-access/delete-snippet.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (salihgueler) [reply src/fragments/lib/datastore/flutter/data-access/delete-snippet.mdx:11]: Since one future relies on the other one and also for the sake of explaining better, I would rather keep the code as it is but improve the readability with comment and stating what @HuiSF mentioned

File: src/fragments/lib/datastore/flutter/getting-started/60_initDataStore.mdx [hunk 9]
```diff
@@ -23,27 +9,39 @@ import 'amplifyconfiguration.dart';
...
-    void initState() {
-        super.initState();
-        _configureAmplify();
+  @override
+  void initState() {
+    super.initState();
+    _configureAmplify();
+  }
+
+  void _configureAmplify() async {
```
[the remark this answers is not in the document]
Comment (salihgueler) [reply src/fragments/lib/datastore/flutter/getting-started/60_initDataStore.mdx:25]: I had two of these slipped out of my eyes :D

File: src/fragments/lib/datastore/flutter/sync/50-selectiveSync.mdx [hunk 10]
```diff
@@ -155,24 +164,30 @@ E.g., `User.LASTNAME.eq("Doe").and(User.CREATEDAT.gt("2020-10-10"))`.
...
+    modelProvider: ModelProvider.instance,
+    syncExpressions: [
+      DataStoreSyncExpression(
+        User.classType,
+        () => User.LASTNAME.eq("Doe").and(User.CREATEDAT.gt("2020-10-10")),
+      ),
+    ],
+  );
+
+  await Amplify.addPlugins([singleEqualsStore, chainedEqualGtStore]);
```
[the remark this answers is not in the document]
Comment (salihgueler) [reply src/fragments/lib/datastore/flutter/sync/50-selectiveSync.mdx:191]: Actually this was an either or expression, I was trying out the possibility. good catch!

File: src/fragments/lib/graphqlapi/flutter/advanced-workflows/30_nested.mdx [hunk 11]
```diff
@@ -19,11 +19,14 @@ final getPostRequest = GraphQLRequest<Post>(
     document: graphQLDocument,
     modelType: Post.classType,
     variables: <String, String>{'id': somePostId},
```
[the remark this answers is not in the document]
Comment (salihgueler) [reply src/fragments/lib/graphqlapi/flutter/advanced-workflows/30_nested.mdx:21]: The somePostId is not defined anywhere above, so for emphasizing the type of it, I think it is valuable

File: src/fragments/lib/graphqlapi/flutter/advanced-workflows/40_multiple.mdx [hunk 12]
```diff
@@ -20,19 +20,25 @@ final multiOperationRequest = GraphQLRequest<String>(
...
-  final todo = Todo.fromJson(jsonData[getTodo]);
+// Do not forget to add this to imports for json.decode
+import 'dart:convert';
+
+...
+
+Future<void> queryMultiOperationRequest(GraphQLRequest operation) async {
+  final response = await Amplify.API.query(request: multiOperationRequest).response;
+  if (response.data != null) {
+    Map<String, dynamic> jsonData = json.decode(response.data);
```
[the remark this answers is not in the document]
Comment (salihgueler) [reply src/fragments/lib/graphqlapi/flutter/advanced-workflows/40_multiple.mdx:39]: I totally agree. There are couple of these got away from me, luckily we have another hunter 😄

File: src/fragments/lib/graphqlapi/flutter/query-data.mdx [hunk 13]
```diff
@@ -44,12 +47,16 @@ For large data sets, you'll need to paginate through the results. After receivin
...
-  final secondReq = firstPageData!.requestForNextResult;
-  final secondRes = await Amplify.API.query(request: secondReq!).response;
+
+Future<List<Todo?>> queryPaginatedListItems() async {
+  final firstRequest = ModelQueries.list<Todo>(Todo.classType, limit: limit);
+  final firstResult = await Amplify.API.query(request: firstRequest).response;
+  final firstPageData = firstResult.data;
+
+  // Indicates there are > 100 todos and we can get the request for the next set.
+  if (firstPageData?.hasNextResult == true) {
```
[the remark this answers is not in the document]
Comment (salihgueler) [reply src/fragments/lib/graphqlapi/flutter/query-data.mdx:57]: I don't have strong opinions on that one so let's carry on with this 👍🏼

File: src/fragments/lib/project-setup/flutter/escape-hatch/escape-hatch.mdx [hunk 14]
```diff
@@ -47,32 +49,36 @@ const AWSSigV4Signer signer = AWSSigV4Signer(
...
+      AWSHeaders.target: 'AWSCognitoIdentityProviderService.DescribeUserPool',
+      AWSHeaders.contentType: 'application/x-amz-json-1.1',
+    },
+    body: json.encode({
+      'UserPoolId': '<YOUR-USER-POOL-ID>',
+    }).codeUnits,
+  );
+
+  // Sign and send the HTTP request
+  final AWSSignedRequest signedRequest = await signer.sign(
```
[the remark this answers is not in the document]
Comment (salihgueler) [reply src/fragments/lib/project-setup/flutter/escape-hatch/escape-hatch.mdx:72]: I remembered you wanted to keep these from over conversation another time but maybe I am wrong :)

File: src/fragments/lib/analytics/flutter/getting-started/20_installLib.mdx [hunk 15]
```diff
@@ -1,18 +1,18 @@
...
 environment:
-  sdk: ">=2.12.0 <3.0.0"
+  sdk: ">=2.17.3 <3.0.0"
 
 dependencies:
 
-  # Should already be added during Project Setup walkthrough 
-  amplify_flutter: ^0.5.0
+  # Should already be added during Project Setup walkthrough
+  amplify_flutter: ^0.5.1
```
[the remark this answers is not in the document]
Comment (abdallahshaban557) [reply src/fragments/lib/analytics/flutter/getting-started/20_installLib.mdx]: I agree - we should just keep this to 0.6.0 for our projects. @salihgueler can we update this everywhere so that with semantic versioning developers are getting our latest and greatest?

File: src/fragments/lib/project-setup/flutter/prereq/prereq.mdx [hunk 16]
```diff
@@ -1,20 +1,2 @@
 - Install [Flutter](https://flutter.dev/docs/get-started/install) version 2.0.0 or higher (make sure you are using a stable version of flutter)
 - Setup your [IDE](https://flutter.dev/docs/get-started/editor?tab=androidstudio)
-- Android API level 21 (Android 5.0) or above
-- iOS platform version of at least 11.0.
-
-<Callout warning>
```
Comment (abdallahshaban557) [inline src/fragments/lib/project-setup/flutter/prereq/prereq.mdx:6]: @salihgueler - do we no longer need this because of Flutter 3? I remember we ran into this issue when developers had to upgrade their apps from versions below 2.10

File: src/fragments/start/getting-started/flutter/setup.mdx [hunk 17]
```diff
@@ -10,20 +10,20 @@ flutter create amplified_todo
...
-      sdk: ">=2.12.0 <3.0.0"
+      sdk: ">=2.17.3 <3.0.0"
 
     dependencies:
       flutter:
         sdk: flutter
 
-      amplify_flutter: ^0.5.0
-      amplify_datastore: ^0.5.0
+      amplify_flutter: ^0.5.1
```
Comment (abdallahshaban557) [inline src/fragments/start/getting-started/flutter/setup.mdx]: same comment about using v 0.6.0 @salihgueler

File: src/fragments/lib/analytics/flutter/getting-started/10_preReq.mdx [hunk 18]
```diff
@@ -1,7 +1,5 @@
 * [Install and configure Amplify CLI](https://docs.amplify.aws/cli/start/install)
 * A Flutter application targeting Flutter SDK >= 2.0.0 (stable version) with Amplify libraries integrated
-    * An iOS configuration targeting at least iOS 11.0
```
Comment (abdallahshaban557) [inline src/fragments/lib/analytics/flutter/getting-started/10_preReq.mdx:3]: I feel like it doesn't make sense to remove those for now. Especially since we have a difference between using these minimum target platforms between categories. Also, because developers might choose to directly go to a category and use it, rather than go through the setup. Same applies for Android here.

File: src/fragments/lib/analytics/flutter/getting-started/12_amplifyConfig.mdx [hunk 19]
```diff
@@ -1 +1,9 @@
-Upon completion, `amplifyconfiguration.dart` should be updated to reference provisioned backend analytics resources. Note that these files should already be a part of your project if you followed the [Project setup walkthrough](/lib/project-setup/create-application).
+Upon completion, you can see the result as the following that is going to be providing you a link to reach your analytics data:
+
+```console
+✔ All resources are updated in the cloud
+
+Pinpoint URL to track events https://<YOUR-PROJECT-RELATED-URL>/analytics/overview
+```
+
+In addition, `amplifyconfiguration.dart` should be updated to reference provisioned backend analytics resources. Note that these files should already be a part of your project if you followed the [Project setup walkthrough](/lib/project-setup/create-application).
```
Comment (abdallahshaban557) [inline src/fragments/lib/analytics/flutter/getting-started/12_amplifyConfig.mdx:9]: Nit: "`amplifyconfiguration.dart` **will** be updated to reference provisioned backend...."

File: src/fragments/lib/analytics/flutter/identifyuser.mdx [hunk 20]
```diff
@@ -1,27 +1,37 @@
 This call sends information that you have specified about a user to Amazon Pinpoint. This could be for an unauthenticated (guest) or an authenticated user.
 
-You can get the current user's ID from the Amplify Auth category as shown below. Be sure you have it added and setup per the Auth category documentation.
+You can get the current user's ID from the Amplify Auth category as shown per the Auth category documentation. Be sure to have it before you set it below.
```
Comment (abdallahshaban557) [inline src/fragments/lib/analytics/flutter/identifyuser.mdx:3]: Nit: be sure to have it setup before you use it as shown below. (And then we can link to getting started with Auth). @salihgueler

File: src/fragments/lib/datastore/flutter/sync/10-installPlugin.mdx [hunk 21]
```diff
@@ -21,11 +23,20 @@ import 'amplifyconfiguration.dart';
 import 'models/ModelProvider.dart';
 ```
 
+Then update your configure function by adding `AmplifyAPI` like mentioned. On earlier implementation configuration was used with `_configureAmplify` function and it can be updated there:
```
Comment (abdallahshaban557) [inline src/fragments/lib/datastore/flutter/sync/10-installPlugin.mdx:26]: nit: can we specify the "function" name here? I think we are super specific about that throughout the getting started guide @salihgueler

File: src/fragments/lib/storage/flutter/getting-started/10_preReq.mdx [hunk 22]
```diff
@@ -1,10 +1,8 @@
 * A Flutter application targeting Flutter SDK >= 2.0.0 (stable version) with Amplify libraries integrated
-    * An iOS configuration targeting at least iOS 11.0
```
Comment (abdallahshaban557) [inline src/fragments/lib/storage/flutter/getting-started/10_preReq.mdx:2]: let us not remove these

File: src/fragments/lib/analytics/flutter/getting-started/20_installLib.mdx (hunk 15 above)
[the remark this answers is not in the document]
Comment (salihgueler) [reply src/fragments/lib/analytics/flutter/getting-started/20_installLib.mdx]: This is already updated. This code is marked as outdated.

File: src/fragments/lib/project-setup/flutter/prereq/prereq.mdx (hunk 16 above)
Comment (salihgueler) [reply src/fragments/lib/project-setup/flutter/prereq/prereq.mdx:6]: I removed these because we already set it up in the getting started part of the libraries. It felt repetetive. Also these did not show how the increase these values, the guide shows them.

File: src/fragments/start/getting-started/flutter/setup.mdx (hunk 17 above)
Comment (salihgueler) [reply src/fragments/start/getting-started/flutter/setup.mdx]: This is also updated already.

File: src/fragments/start/getting-started/flutter/setup.mdx [hunk 23]
```diff
@@ -54,4 +54,8 @@ From your project root, navigate to the `android/app/` directory and modify `bui
 minSdkVersion 21
 ```
 
+import flutter1 from '/src/fragments/start/getting-started/flutter/prereq.mdx';
```
[the remark this answers is not in the document]
Comment (salihgueler) [reply src/fragments/start/getting-started/flutter/setup.mdx:57]: Yes I think so, otherwise it feels so separated.

File: src/fragments/lib/analytics/flutter/getting-started/10_preReq.mdx (hunk 18 above)
Comment (salihgueler) [reply src/fragments/lib/analytics/flutter/getting-started/10_preReq.mdx:3]: I can always revert them if you think these are valuable. 👍🏼

File: src/fragments/lib/project-setup/flutter/escape-hatch/escape-hatch.mdx [hunk 24]
```diff
@@ -1,25 +1,27 @@
...
 ### Using the Signer
 
-To get started using the signer, add it as a dependency in your `pubspec.yaml` and create an instance in your project.
+To get started using the signer, add it as a dependency in your `pubspec.yaml` like the following:
 
 ```yaml
 dependencies:
-  aws_common: ^0.1.0
-  aws_signature_v4: ^0.1.0
+  aws_common: ^0.1.1
```
[the remark this answers is not in the document]
Comment (salihgueler) [reply src/fragments/lib/project-setup/flutter/escape-hatch/escape-hatch.mdx:10]: These are also updated to their latest versions.

File: src/fragments/lib/datastore/flutter/sync/10-installPlugin.mdx (hunk 21 above)
Comment (salihgueler) [reply src/fragments/lib/datastore/flutter/sync/10-installPlugin.mdx:26]: I already mention it actually but this comment means it is not clear enough, let me rephrase myself :)

File: src/fragments/lib/project-setup/flutter/escape-hatch/escape-hatch.mdx (hunk 24 above)
[the remark this answers is not in the document]
Comment (salihgueler) [reply src/fragments/lib/project-setup/flutter/escape-hatch/escape-hatch.mdx:10]: I will revert these but I think we need to talk about it. I think this version problem is sending wrong signals. Maybe we should tell them to add libraries with flutter CLI tools:

```
flutter pub add aws_common
```

WDYT?

**Review by abdallahshaban557 — APPROVED** [review]
LGTM! Thank you so much @salihgueler!
